### DISCUSSION TITLE
What does the `set_typ_vals_ode()` function do?

### DISCUSSION THREAD

**Discussion by ensonun** [AMReX-Combustion/PelePhysics#532] [Q&A] [answered]
Hi,

I was wondering what the `set_typ_vals_ode()` function really does.

I see that it sets up the `ReactorBase::m_typ_vals` array, which is passed to the `utils::set_sundials_solver_tols()` function, which then is used to set a `ratol` array. However, I don't see the `ratol` array being used anywhere else. 

I was a bit confused about the reason behind this. Or is it some legacy configuration that is no longer used?

I appreciate your help. Thank you!

**Comment by baperry2** [comment, marked as answer]
This is not deprecated legacy code. `set_typ_vals_ode` specifies a characteristic value for each species in the chemical system, which is then used in setting absolute error tolerances for the Sundials integrators. This is can be important because the orders of magnitudes for species mass fractions can range from O(1) for major species to very very tiny for radicals, so the appropriate absolute tolerance varies from species to species. 

You can see where the tolerances are set for CVode or other solvers here: https://github.com/AMReX-Combustion/PelePhysics/blob/e371e15e609195265b9b82e9c2f464bff400ccfe/Source/Reactions/ReactorUtils.H#L148.

**Reply by ensonun** [reply]
Oh I see. `ratol` is the pointer to atol. Sorry I missed that. Thank you so much for the explanation.
